pkg/rpm/resolver/templates/prepare-tarball.sh.tpl

```diff
@@ -33,7 +33,23 @@ if [ `wc -w <<< $RAW_FILE` -ne 1 ]; then
 	exit 1
 fi
 
-virt-tar-out -a $RAW_FILE / - | pigz --best > $WORK_DIR/{{.ArchiveName}}
+# Test the block size of the base image and adapt to suit either 512/4096 byte images
+BLOCKSIZE=512
+if ! guestfish -i --blocksize=$BLOCKSIZE -a $RAW_FILE echo "[INFO] 512 byte sector check successful."; then
+        echo "[WARN] Failed to access image with 512 byte sector size, trying 4096 bytes."
+        BLOCKSIZE=4096
```

Comment:
Shall we do the same check here instead of assuming 4096?

Reply:
@e-minguez as per the previous comment, this is the check. This then feeds into the next call `virt-tar-out` which needs to also be aware of the block size. It will use 4096 *only* if a simple echo check with 512 fails.